### PR TITLE
GCLOUD_PROJECT -> GOOGLE_CLOUD_PROJECT

### PR DESCRIPTION
**Pull request by grant**
We [prefer](https://cloud.google.com/functions/docs/env-var#reserved_keys_key_validation) `GOOGLE_CLOUD_PROJECT` over `GCLOUD_PROJECT`. Rename a such.

See various docs and comment threads about this:
- https://cloud.google.com/appengine/docs/flexible/nodejs/runtime#environment_variables
- https://github.com/googleapis/nodejs-common/issues/20#issuecomment-360209232
- https://github.com/googleapis/google-cloud-php/issues/827

### CODE REVIEW

**Comment by fhinkel** [comment]
cc @steren

**Comment by fhinkel** [comment]
Is this for https://github.com/GoogleCloudPlatform/nodejs-getting-started/issues/251?

**Comment by grant** [comment]
Any update on reviewing this PR?

**Comment by grant** [comment]
Ping~
Is there anything specific I can do to help the PR get reviewed faster?

File: 2-structured-data/books/model-datastore.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ const config = require('../config');
 
 // [START config]
 const ds = Datastore({
-  projectId: config.get('GCLOUD_PROJECT'),
+  projectId: config.get('GOOGLE_CLOUD_PROJECT'),
```
Comment (steren) [inline 2-structured-data/books/model-datastore.js]: Why do you need to set the project ID here?
The library is supposed to take it from the metadata server or from the GOOGLE_APPLICATION_CREDENTIAL context

File: 2-structured-data/config.js [hunk 2]
```diff
@@ -51,7 +51,7 @@ nconf
   });
 
 // Check for required settings
-checkConfig('GCLOUD_PROJECT');
+checkConfig('GOOGLE_CLOUD_PROJECT');
```
Comment (steren) [inline 2-structured-data/config.js]: Why is the project ID a required config? 

The code is supposed to take it from the metadata server or from the GOOGLE_APPLICATION_CREDENTIAL context.

File: 3-binary-data/books/model-datastore.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ const Datastore = require('@google-cloud/datastore');
 const config = require('../config');
 
 const ds = Datastore({
-  projectId: config.get('GCLOUD_PROJECT'),
+  projectId: config.get('GOOGLE_CLOUD_PROJECT'),
```
Comment (steren) [inline 3-binary-data/books/model-datastore.js]: Why is the project ID a required config? 

The code is supposed to take it from the metadata server or from the GOOGLE_APPLICATION_CREDENTIAL context.

File: 3-binary-data/config.js [hunk 4]
```diff
@@ -55,7 +55,7 @@ nconf
   });
 
 // Check for required settings
-checkConfig('GCLOUD_PROJECT');
+checkConfig('GOOGLE_CLOUD_PROJECT');
```
Comment (steren) [inline 3-binary-data/config.js]: same remark here

File: 4-auth/books/model-datastore.js [hunk 5]
```diff
@@ -17,7 +17,7 @@ const Datastore = require('@google-cloud/datastore');
 const config = require('../config');
 
 const ds = Datastore({
-  projectId: config.get('GCLOUD_PROJECT'),
+  projectId: config.get('GOOGLE_CLOUD_PROJECT'),
```
Comment (steren) [inline 4-auth/books/model-datastore.js]: Why is the project ID a required config? 

The code is supposed to take it from the metadata server or from the GOOGLE_APPLICATION_CREDENTIAL context.

File: 7-gce/README.md [hunk 6]
```diff
@@ -43,7 +43,7 @@ and deploying this sample.
 
         cp config-default.json config.json
 
-    * Set `GCLOUD_PROJECT` in `config.json` to your Google Cloud Platform
+    * Set `GOOGLE_CLOUD_PROJECT` in `config.json` to your Google Cloud Platform
```
Comment (steren) [inline 7-gce/README.md]: I do not understand why users need to do that.

**Comment by fhinkel** [comment]
+1 to Steren's comments. Ping me when it's ready for another review. Sorry that I didn't see that, too many GitHub issues :(

**Comment by grant** [comment]
@steren 
The goal of this PR is to remove the legacy key by replacing it with the new key.

I'm not really at the point of familiarity with these samples of questioning why the key is set in the first place for each case. I think it would be reasonable to merge a PR that swaps the keys to not have users set legacy env vars.

What would you like me to do?

**Comment by steren** [comment]
The big issue to fix is to remove the dependency on the env var used to get the current project. Our client libraries *do not* need to get the Project ID in their constructor when then run on GCP (*anywhere* on GCP). And locally, the project ID is derived from 
`GOOGLE_APPLICATION_CREDENTIALS`

I am OK for a first CL that just swap the string, but this is just masking the underlying issue.

**Comment by grant** [comment]
I propose that we first merge this PR in an effort to fix part of #251.
And removing the env var in places where it can be removed would be a second step for #251.

Otherwise, let's close this PR.

**Comment by steren** [comment]
LGTM, but please do not mark #251 as fixed.

**Comment by grant** [comment]
I've removed references to fixing #251 and I've updated the branch. Please take a look.

**Comment by fhinkel** [comment]
I don't think there's any benefit in renaming the variable. I'm opening a PR that gets rid of setting the project.

**Comment by grant** [comment]
Well, I think there's value in not using deprecated settings in code samples...

**Comment by murfett-au** [comment]
I have several use cases where we need access to the GCP project id. We use it to generate the name of a artifact repository, and in a call to `SecretManagerClient.accessSecretVersion({ name:'projects/'+projectId+'/secrets/..`. The projectId value is different in each of our environments. I would prefer not to find the projectId in process.env, rather than needing to await an async function.

**Pull request closed without merging**